### DISCUSSION TITLE
Does gorilla support Java? If so, does it support custom Java API?

### DISCUSSION THREAD

**Discussion by romiIntellinum** [ShishirPatil/gorilla#67] [Q&A] [answered]
Hi,

Currently gorilla API examples are for python API. Does gorilla support Java? 
Can gorilla be trained (or fine tuned) to understand custom Java library? 

Thank you,
RH

**Comment by lgersman** [comment]
I have a similar question. I would like to use gorilla to create http rest calls (similarly to what I can do with langchain's openapi chain, but faster/cheaper 😅). And last but not least to compute javascript calls out if human language for use in the browser.

Do you plan to integrate more than just python apis ?

**Comment by ShishirPatil** [comment, marked as answer]
Thanks for your interest @romiIntellinum and @lgersman Yes, indeed. There is nothing in-particular about python that we care about. In-fact for [gorilla-cli](https://github.com/gorilla-llm/gorilla-cli) we trained a model with only BASH commands. So, yes! Custom restful API calls and custom Java APIs is definitely a possibility!

**Comment by lgersman** [comment]
Are the gorilla-cli training data publicly available ?

Shishir Patil ***@***.***> schrieb am Do., 27. Juli 2023,
15:27:
